### DISCUSSION TITLE
Can I change the colour of a button using streamdeckc?

### DISCUSSION THREAD

**Discussion by proffalken** [streamdeck-linux-gui/streamdeck-linux-gui#192] [Q&A]
Pretty much the title.

I've really started to depend on `streamdeckc` to update states on my streamdeck, but the one thing I don't appear to be able to do is change the background colour of a button.

My use-case is a script that runs when a button is pressed.  By default, the button background is RED to tell me that there are changes that need to be committed, and pressing the button should turn the background GREEN so I know it's been done.

I could look at using different icons for this, but I'd rather keep the icon and just change the background colour.

Is there any way to do this?

**Comment by Quintar** [comment]
Not really streamdeckdc has as of yet no ability to change the backgroundcolor of a button. I'd use a neatly sized png.

**Comment by proffalken** [comment]
OK, thanks - I assume it wouldn't be too hard to add given that you can change it from the GUI?

Happy to give it a go, from what I can tell from the code it would be a case of adding a command to https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/blob/main/streamdeck_ui/cli/commands.py and then calling the code in https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/blob/main/streamdeck_ui/display/background_color_filter.py ?

**Comment by dhtseany** [comment]
I for one would love this ability, I have 5 states to possibly change to
across 20 buttons and anytime I add a new button I have to make a ton of
icons to match. Simply changing the background color via streamdeckc would
be a drastic improvement to my workflow. I hope you succeed, Matthew!
